datapipelines/src/main/kotlin/io/customer/datapipelines/migration/TrackingMigrationProcessor.kt

```diff
@@ -123,7 +123,7 @@ internal class TrackingMigrationProcessor(
         }
 
         trackEvent.timestamp = task.timestamp.toString()
-        task.identifier?.let { trackEvent.userId = it }
+        trackEvent.userId = task.identifier
```

Comment:
[iOS uses `anonymousId` today](https://github.com/customerio/customerio-ios/blob/2f4e4e4d07bedffc251b0f0a19b3e2bd55b0375d/Sources/DataPipeline/DataPipelineImplementation.swift#L355), we can do the same by updating it as something like follows:

```kotlin
if ((trackEvent as? TrackEvent)?.event == EventNames.METRIC_DELIVERY) {
    trackEvent.anonymousId = task.identifier
} else {
    trackEvent.userId = task.identifier
}
```

But I think setting `userId` should work for us and keep things simple.